### ISSUE TITLE
`Expected type in class pattern; found "Any"` for ignored missing import in match/case

### ISSUE THREAD

**Issue by ds-cbo** [python/mypy#14477]
**To Reproduce**

```python
from somewhere import Foo

match None:
    case Foo():
        ...
```

https://mypy-play.net/?mypy=latest&python=3.11&flags=ignore-missing-imports&gist=03694364e4e855575f23ba9c625e804e

**Expected Behavior**

no error

**Actual Behavior**

```
$ mypy --ignore-missing-imports test.py
test.py:4: error: Expected type in class pattern; found "Any"  [misc]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

mypy 0.991 without config files on python 3.10 and 3.11

**Comment by tinkerware** [comment]
This can be an occasional stumbling block that's quite difficult to find a way out of without the right error message. In my case, I ran into two internal libraries, `A` and `B`, where `A` had imports from `B`. A also had the `ignore_missing_imports = true` setting to deal with some legacy 3rd-party libraries.

Code that looks like following fails in this set-up with the error in this issue:

```python
from B import Foo, Bar

def do_stuff(value: Foo | Bar) → None:
  match value:
    case Foo(a=a):
      ...
    case Bar(b=b):
      ...
```

No error is given if library `B` adds the `py.typed` marker. At the very least, a more descriptive error with a fix suggestion seems useful. Using `ignore_missing_imports = false` does give the user a hint with the `skipping analyzing ...` error.
